### PR TITLE
initial commit for DB HW week2

### PR DESCRIPTION
**Pull request by agkabir**

### CODE REVIEW

File: databases/week2/part1.sql [hunk 1]
```diff
@@ -0,0 +1,18 @@
+USE hyf_class2023;
+-- add a task
+insert into task (title, description, created, updated, due_date, status_id, user_id) values ('DB week 2 home work', 'Insert a new task', '2017-10-20 01:41:53', '2017-10-04 07:19:56', '2017-12-23 10:13:42', 2, NULL);
```
Comment (Elia-Git) [inline databases/week2/part1.sql:3]: I don't think you are able to insert this data into the task table as `user_id` cannot be `NULL`. Please check the query from week 1 we use to create a user table.

File: databases/week2/part1.sql [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+SELECT * FROM task WHERE id = 36;
+-- Change a task status
+UPDATE task SET status_id = 1 WHERE id = 36;
+SELECT * FROM task WHERE id = 36;
+-- Mark a task as complete (have to set status id = 3 )
+UPDATE task SET status_id = 3 WHERE id = 36;
+SELECT * FROM task WHERE id = 36;
+-- Delete a task
+DELETE from task WHERE id = 36;
+SELECT * FROM task;
```
Comment (Elia-Git) [inline databases/week2/part1.sql:18]: good job here!

File: databases/week2/create_database.sql [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  `email` varchar(255) NOT NULL,
+  `phone` varchar(255) NULL,
+  PRIMARY KEY (`id`)
+) ENGINE=InnoDB;
+
+CREATE TABLE IF NOT EXISTS candidate (
+  `id` int(10) unsigned NOT NULL AUTO_INCREMENT,
+  `name` varchar(255) NOT NULL,
+  `email` varchar(255) NOT NULL,
+  `phone` varchar(255) NOT NULL,
```
Comment (Elia-Git) [inline databases/week2/create_database.sql:30]: Wondering to know WHY you want to assign 255 chars for a human name, email, or phone number(they can't be that much longer). Applies to the above too.

File: databases/week2/part2.sql [hunk 4]
```diff
@@ -0,0 +1,25 @@
+USE hyf_class2023;
+
+-- create table Class with the columns: id, name, begins (date), ends (date)
+CREATE TABLE IF NOT EXISTS Class (
+  `id` int(10) unsigned NOT NULL AUTO_INCREMENT,
+  `name` VARCHAR(255) NOT NULL,
+  `begins_date` DATETIME NOT NULL,
+  `ends_date` DATETIME NULL DEFAULT NULL,
```
Comment (Elia-Git) [inline databases/week2/part2.sql:8]: columns` begins` &` ends` were supposed to have date not datetime as a datatype.

File: databases/week2/part3.sql [hunk 5]
```diff
@@ -0,0 +1,15 @@
+USE hyf_class2023;
+-- Get all the tasks assigned to users whose email ends in @spotify.com
+SELECT task.id, task.title, user.email FROM task INNER JOIN user ON task.user_id = user.id WHERE user.email LIKE '%@spotify.com';
+-- Get all the tasks for 'Donald Duck' with status 'Not started'
+SELECT task.id, task.title, task.created, user.name FROM task
+INNER JOIN status ON task.status_id = status.id
+INNER JOIN user ON task.user_id = user.id
+WHERE status.name='Not started' AND user.name='Donald Duck';
```
Comment (Elia-Git) [inline databases/week2/part3.sql:8]: Please compare your & see the diff
```
SELECT task.id, task.title, task.created, user.name 
FROM task
join user_task on task.id=user_task.task_id
JOIN status ON task.status_id = status.id 
JOIN user ON user_task.user_id = user.id
WHERE status.name='Not started' AND user.name='Donald Duck';
```

File: databases/week2/part3.sql [hunk 6]
```diff
@@ -0,0 +1,15 @@
+USE hyf_class2023;
+-- Get all the tasks assigned to users whose email ends in @spotify.com
+SELECT task.id, task.title, user.email FROM task INNER JOIN user ON task.user_id = user.id WHERE user.email LIKE '%@spotify.com';
```
Comment (Elia-Git) [inline databases/week2/part3.sql:3]: Your query is only giving out 1 task where as the user has 4 tasks assigned. My suggestion would be something like this...
```
SELECT user.name,user.email, task.title,task.id
FROM task
JOIN user_task ON task.id = user_task.task_id
JOIN user ON user.id = user_task.user_id
WHERE user.email LIKE '%@spotify.com';
```

File: databases/week2/part3.sql [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+INNER JOIN status ON task.status_id = status.id
+INNER JOIN user ON task.user_id = user.id
+WHERE status.name='Not started' AND user.name='Donald Duck';
+-- Get all the tasks for 'Maryrose Meadows' that were created in september (hint: month(created)=month_number)
+SELECT task.id, task.title, task.created, user.name FROM task
+INNER JOIN user ON task.user_id = user.id
+WHERE MONTH(task.created)=9 AND user.name='Maryrose Meadows';
+-- Find how many tasks where created in each month, e.g. how many tasks were created in october, how many tasks were created in november, etc.
+SELECT MONTH(created) as 'Month Number', count(id) as 'No of tasks by month'
+FROM task GROUP BY MONTH(created);
```
Comment (Elia-Git) [inline databases/week2/part3.sql:15]: A small fix by using MONTHNAME() function...test this pls, so you can see how many tasks created which month 
```
SELECT MONTHNAME(created) as 'Month', count(id) as 'No of tasks by month'
FROM task GROUP BY month;
```

File: databases/week2/part3.sql [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+SELECT task.id, task.title, user.email FROM task INNER JOIN user ON task.user_id = user.id WHERE user.email LIKE '%@spotify.com';
+-- Get all the tasks for 'Donald Duck' with status 'Not started'
+SELECT task.id, task.title, task.created, user.name FROM task
+INNER JOIN status ON task.status_id = status.id
+INNER JOIN user ON task.user_id = user.id
+WHERE status.name='Not started' AND user.name='Donald Duck';
+-- Get all the tasks for 'Maryrose Meadows' that were created in september (hint: month(created)=month_number)
+SELECT task.id, task.title, task.created, user.name FROM task
+INNER JOIN user ON task.user_id = user.id
+WHERE MONTH(task.created)=9 AND user.name='Maryrose Meadows';
```
Comment (Elia-Git) [inline databases/week2/part3.sql:12]: Pls, be aware you are not using table `user_task`  where necessary & your query doesn't make any result...
```
SELECT task.id,user.name, task.title, task.created
FROM task
JOIN user_task ON task.id = user_task.task_id
JOIN user  ON user.id = user_task.user_id
WHERE user.name LIKE 'Maryrose Meadows' AND MONTH(created)=09;
```

**Review by Elia-Git — CHANGES_REQUESTED** [review]
Good job!
Please consider the given comments for small fix.
Thx.

File: databases/week2/part1.sql (hunk 1 above)
Comment (agkabir) [reply databases/week2/part1.sql:3]: You are right but I think I worked with a table where user_id accept null.

File: databases/week2/create_database.sql (hunk 3 above)
Comment (agkabir) [reply databases/week2/create_database.sql:30]: yes it could be in between 50 to 100 characters.

File: databases/week2/part2.sql (hunk 4 above)
Comment (agkabir) [reply databases/week2/part2.sql:8]: ok.

File: databases/week2/part3.sql (hunk 5 above)
Comment (agkabir) [reply databases/week2/part3.sql:8]: it is my bad, I worked with the table I created for lesson 1 where I did not have user_task table

File: databases/week2/part3.sql (hunk 6 above)
Comment (agkabir) [reply databases/week2/part3.sql:3]: same here

**Pull request merged**